<?xml version="1.0" ?>
<robot name="robot_model_v2">
  <joint name="front_wheel_joint" type="continuous">
    <parent link="base_link"/>
    <child link="front_wheel_link"/>
    <origin rpy="0 0 0" xyz="2.0000e-05 -1.5394e-01 -2.3920e-02"/>
    <axis xyz="1 0 0"/>
  </joint>
  <joint name="left_motor_joint" type="fixed">
    <parent link="base_link"/>
    <child link="left_motor_link"/>
    <origin rpy="0 0 0" xyz="0.1025   0.09107 -0.01392"/>
    <axis xyz="0 0 0"/>
    <limit effort="0" lower="0" upper="0" velocity="0"/>
  </joint>
  <joint name="right_motor_joint" type="fixed">
    <parent link="base_link"/>
    <child link="right_motor_link"/>
    <origin rpy="0 0 0" xyz="-0.1025   0.09107 -0.01392"/>
    <axis xyz="0 0 0"/>
    <limit effort="0" lower="0" upper="0" velocity="0"/>
  </joint>
  <joint name="left_wheel_joint" type="continuous">
    <parent link="left_motor_link"/>
    <child link="left_wheel_link"/>
    <origin rpy="1.5708 0     1.5708" xyz="0.045  0    -0.01"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="right_wheel_joint" type="continuous">
    <parent link="right_motor_link"/>
    <child link="right_wheel_link"/>
    <origin rpy="1.5708 0     1.5708" xyz="-0.045  0    -0.01"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="base_link_joint" type="fixed">
    <parent link="base_footprint"/>
    <child link="base_link"/>
    <origin rpy="0 0 1.5707" xyz="0 0 0"/>
    <axis xyz="0 0 0"/>
    <limit effort="0" lower="0" upper="0" velocity="0"/>
  </joint>
  <joint name="laser_link_joint" type="fixed">
    <parent link="base_link"/>
    <child link="base_scan"/>
    <origin rpy="0 0 1.5707" xyz="0 0 0"/>
    <axis xyz="0 0 0"/>
    <limit effort="0" lower="0" upper="0" velocity="0"/>
  </joint>
  <link name="base_footprint"></link>
  <link name="base_scan"></link>
  <link name="base_link">
    <inertial>
      <mass value="1"/>
      <origin rpy="0 0 0" xyz="0      -0.03642  0.31357"/>
      <inertia ixx="0.166667" ixy="0" ixz="0" iyy="0.166667" iyz="0" izz="0.166667"/>
    </inertial>
    <collision name="collision">
      <origin rpy="0 0 0" xyz="0      -0.03642  0.11357"/>
      <geometry>
        <box size="0.255 0.305 0.205"/>
      </geometry>
    </collision>
    <visual name="visual">
      <origin rpy="0 0 0" xyz="0      -0.03642  0.11357"/>
      <geometry>
        <box size="0.255 0.305 0.205"/>
      </geometry>
    </visual>
  </link>
  <link name="left_motor_link">
    <inertial>
      <mass value="1"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0.166667" ixy="0" ixz="0" iyy="0.166667" iyz="0" izz="0.166667"/>
    </inertial>
    <collision name="collision">
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.05 0.05 0.05"/>
      </geometry>
    </collision>
    <visual name="visual">
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.05 0.05 0.05"/>
      </geometry>
    </visual>
  </link>
  <link name="right_motor_link">
    <inertial>
      <mass value="1"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0.166667" ixy="0" ixz="0" iyy="0.166667" iyz="0" izz="0.166667"/>
    </inertial>
    <collision name="collision">
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.05 0.05 0.05"/>
      </geometry>
    </collision>
    <visual name="visual">
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.05 0.05 0.05"/>
      </geometry>
    </visual>
  </link>
  <link name="left_wheel_link">
    <inertial>
      <mass value="1"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0.145833" ixy="0" ixz="0" iyy="0.145833" iyz="0" izz="0.125"/>
    </inertial>
    <collision name="collision">
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.04" radius="0.035"/>
      </geometry>
    </collision>
    <visual name="visual">
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.04" radius="0.035"/>
      </geometry>
    </visual>
  </link>
  <link name="right_wheel_link">
    <inertial>
      <mass value="1"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0.145833" ixy="0" ixz="0" iyy="0.145833" iyz="0" izz="0.125"/>
    </inertial>
    <collision name="collision">
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.04" radius="0.035"/>
      </geometry>
    </collision>
    <visual name="visual">
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.04" radius="0.035"/>
      </geometry>
    </visual>
  </link>
  <link name="front_wheel_link">
    <inertial>
      <mass value="1"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0.1" ixy="0" ixz="0" iyy="0.1" iyz="0" izz="0.1"/>
    </inertial>
    <collision name="collision">
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.035"/>
      </geometry>
    </collision>
    <visual name="visual">
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.035"/>
      </geometry>
    </visual>
  </link>
</robot>
